examples/restaurant_demo/README.md

```diff
@@ -0,0 +1,260 @@
+each entry hour.
+
+More generally, a single partition represents a subset of aggregated data
+corresponding to a given value of the aggregation criterion. Graphically, a
+single partition is represented as a bar on the aggregated bar chart.
+
+We say that a visitor *contributes* to a given partition if their data matches
+the partition criterion. For example, if a visitor enters between 8 AM and 9 AM,
+they *contribute* to the *8 AM partition*.
+
+Recall that in the the example above, a visitor can enter the restaurant only
+once per day. This implies three *contribution bounds*:
+
+*   *Maximum partitions contributed*: to how many partitions can a visitor
+    contribute? In our example, a visitor can contribute up to one partition. In
+    other words, there is at most one time-slot when a visitor with a given id
+    can enter the restaurant.
+*   *Maximum contributed value*: what is the maximum value that can be
+    contributed by a visitor to a partition? In our example, we count the number
+    of visits, so the maximum contributed value is simply *1*.
+*   *Maximum contributions per partition*: how many times can a visitor
+    contribute to a partition? In our example, a visitor can contribute to a
+    partition at most once. In other words, a visitor can enter the restaurant
+    only once at a given hour.
+
+Why is this important? Differential Privacy adjusts the amount of noise to mask
+contributions of each visitor. More contributions require more noise.
+
+Next, we analyse for the remaining three cases:
+
+* Count visits by day of week
+* Sum-up revenue per day of the week
+* Sum-up revenue per day of the week with preaggregation
+
+## Count visits by day of week
+
+The previous example made some over-simplifying assumptions. Now, let’s have a
+look at the use-case where visitors can contribute to multiple partitions.
+
+Imagine Alice decides to let visitors know which days are the busiest at her
+restaurant. For this, she calculates how many people visit the restaurant every
+day of the week. For simplicity, let’s assume a visitor enters the restaurant at
+most once a day but multiple times a week.
+
+Visit data for a week is stored in the `week_data.csv` file.
+
+The results are illustrated in the image below.
+
+![Counts per week day](img/counts_per_day.png)
+
+As you can see, the private values slightly differ from the actual ones but the
+overall trend is preserved.
+
+Now, let’s take a closer look at the technical details. Speaking in terms of
+*partitions* and *contributions*, the resulting bar chart has 7 partitions: one
+for each day of the week. A visitor may enter the restaurant once a day and
+hence contribute to a partition at most once. A visitor may enter the restaurant
+several times a week and hence contribute to up to 7 partitions. The code below
+uses `Count` to calculate the differentially private count of visits for a
+single day.
+
+```java
```

Comment:
this is the code snippet from google's example. pls add the one from pydp lib example

Reply:
Done